linode/image/framework_models.go

```diff
@@ -0,0 +1,47 @@
+package image
+
+import (
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+)
+
+// ImageModel describes the Terraform resource data model to match the
+// resource schema.
+type ImageModel struct {
+	ID          types.String `tfsdk:"id"`
+	Label       types.String `tfsdk:"label"`
+	Description types.String `tfsdk:"description"`
+	Created     types.String `tfsdk:"created"`
+	CreatedBy   types.String `tfsdk:"created_by"`
+	Deprecated  types.Bool   `tfsdk:"deprecated"`
+	IsPublic    types.Bool   `tfsdk:"is_public"`
+	Size        types.Int64  `tfsdk:"size"`
+	Status      types.String `tfsdk:"status"`
+	Type        types.String `tfsdk:"type"`
+	Expiry      types.String `tfsdk:"expiry"`
+	Vendor      types.String `tfsdk:"vendor"`
+}
+
+func (data *ImageModel) ParseImage(
+	image *linodego.Image,
+) {
+	data.ID = types.StringValue(image.ID)
+	data.Label = types.StringValue(image.Label)
+
+	data.Description = types.StringValue(image.Description)
+	if image.Created != nil {
+		data.Created = types.StringValue(image.Created.Format(time.RFC3339))
+	}
+	if image.Expiry != nil {
```

Comment:
I think we can set these two values to `types.StringNull()` when the fields are `nil`. Realistically this shouldn't be a case we run into, but it might be worth having that extra redundancy just in case 👍